homeassistant/helpers/debounce.py

```diff
@@ -2,14 +2,19 @@
 from __future__ import annotations
 
 import asyncio
-from collections.abc import Awaitable, Callable
+from collections.abc import Callable
 from logging import Logger
-from typing import Any
+from typing import Generic, TypeVar
+
+from typing_extensions import ParamSpec
 
 from homeassistant.core import HassJob, HomeAssistant, callback
 
+_R_co = TypeVar("_R_co", covariant=True)
+_P = ParamSpec("_P")
+
 
-class Debouncer:
+class Debouncer(Generic[_P, _R_co]):
```

Comment:
The parameter type will always be the same as the debouncer doesn't allow passing parameters to the target. Should we remove the ParamSpec type from the generic type?